spec/plugin_mixin/http_client_ssl_spec.rb

```diff
@@ -0,0 +1,448 @@
+  include LogStash::PluginMixins::HttpClient[:with_deprecated => true]
+  config_name 'with_deprecated'
+end
+
+class PluginWithDeprecatedFalse < LogStash::Inputs::Base
+  include LogStash::PluginMixins::HttpClient[:with_deprecated => false]
+  config_name 'without_deprecated'
+end
+
+describe PluginWithNoModuleConfig do
+  let(:plugin_class) { PluginWithNoModuleConfig }
+
+  it_behaves_like 'a client with deprecated ssl options'
+
+  it 'includes DeprecatedSslConfigSupport module' do
+    expect(plugin_class.ancestors).to include(LogStash::PluginMixins::HttpClient::DeprecatedSslConfigSupport)
+  end
+end
+
+describe PluginWithDeprecatedFalse do
+  let(:plugin_class) { PluginWithDeprecatedFalse }
+
+  it_behaves_like 'a client with standardized ssl options'
+
+  it 'does not include DeprecatedSslConfigSupport module' do
+    expect(plugin_class.ancestors).to_not include(LogStash::PluginMixins::HttpClient::DeprecatedSslConfigSupport)
+  end
+end
+
+describe PluginWithDeprecatedTrue do
+  let(:plugin_class) { PluginWithDeprecatedTrue }
+
+  it_behaves_like 'a client with deprecated ssl options'
+
+  it_behaves_like 'a client with standardized ssl options'
+
+  context 'setting deprecate configs' do
+    let(:cacert) { Stud::Temporary.file.path }
+    let(:client_cert) { Stud::Temporary.file.path }
+    let(:client_key) { Stud::Temporary.file.path }
+    let(:keystore) { Stud::Temporary.file.path }
+    let(:keystore_type) { 'pkcs12' }
+    let(:keystore_password) { 'bar' }
+    let(:truststore) { Stud::Temporary.file.path }
+    let(:truststore_type) { 'pkcs12' }
+    let(:truststore_password) { 'foo' }
+
+    let(:settings) do
+      {
+        'cacert' => cacert,
+        'client_cert' => client_cert,
+        'client_key' => client_key,
+        'keystore' => keystore,
+        'keystore_password' => keystore_password,
+        'keystore_type' => keystore_type,
+        'truststore' => truststore,
+        'truststore_password' => truststore_password,
+        'truststore_type' => truststore_type
+      }
+    end
+
+    subject { plugin_class.new(settings) }
+
+    after do
+      File.unlink(cacert)
+      File.unlink(client_cert)
+      File.unlink(client_key)
+      File.unlink(keystore)
+      File.unlink(truststore)
+    end
+
+    it 'should normalize deprecated settings' do
+      expect(subject.ssl_certificate_authorities).to eq([cacert])
+      expect(subject.ssl_certificate).to eq(client_cert)
+      expect(subject.ssl_key).to eq(client_key)
+      expect(subject.ssl_keystore_path).to eq(keystore)
+      expect(subject.ssl_keystore_password.value).to eq(keystore_password)
+      expect(subject.ssl_keystore_type).to eq(keystore_type)
+      expect(subject.ssl_truststore_path).to eq(truststore)
+      expect(subject.ssl_truststore_password.value).to eq(truststore_password)
+      expect(subject.ssl_truststore_type).to eq(truststore_type)
+
+      expect(subject.params['ssl_certificate_authorities']).to eq([cacert])
```

Comment:
I didn't check the plugins that include this mixin. When I mentioned I saw plugins relying on the `params` values, I meant, for example, the ES output [plugin](https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/main/lib/logstash/outputs/elasticsearch/http_client_builder.rb#L59C1-L59C1), which was the motivator to almost all normalized plugins also inject the normalized values into `params`. 

Anyway, I'm fine changing it if you prefer to not inject those `params` values.

Reply:
Let's not inject them; it adds one more "source of truth" and consumers of the mixin already have enough.

On the ES output side, it would be wise to firm up its API boundary between the plugin itself and the client it uses, but we don't need to leak its messy access pattern into this unrelated project :)

Reply:
Done! :)